rasa_nlu/classifiers/regex_intent_classifier.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+from builtins import map
+from typing import Any
+from typing import Dict
+from typing import Text
+import re
+
+from rasa_nlu.components import Component
+
+
+class RegExIntentClassifier(Component):
+
+    name = "intent_classifier_regex"
+
+    context_provides = {
+        "process": ["intent"],
+    }
+
+    output_provides = ["intent"]
+
+    def __init__(self, clf=None, regex_dict=None):
+        self.clf = clf
+        self.regex_dict = {}
+
+    def train(self, training_data):
```

Comment:
The incoming `trianing_data` parameter is of type `TrainingData ` (can be found in `training_data.py`). Before the data gets passed through the components in the pipeline the data will be read here: https://github.com/RasaHQ/rasa_nlu/blob/master/rasa_nlu/converters.py#L205

That function loads the different parts of the training data file, e.g. the `common_examples` section from the training data file (e.g. as seen in https://github.com/RasaHQ/rasa_nlu/blob/master/data/examples/rasa/demo-rasa.json#L3).

The regex expressions should be put into the training data file (`demo-rasa.json`) and they should be read within the `load_rasa_data` in the `converters.py`. The read regex expressions should be passed to the TrainingData object https://github.com/RasaHQ/rasa_nlu/blob/master/rasa_nlu/training_data.py#L33 and stored. You can then access these stored regexes in this function.